conan/cli/commands/remove.py

```diff
@@ -91,7 +91,7 @@ def confirmation(message):
         if packages is None:
             if confirmation(f"Remove the recipe and all the packages of '{ref.repr_notime()}'?"):
                 conan_api.remove.recipe(ref, remote=remote)
-            else:
+            elif not args.dry_run:
```

Comment:
I think this reads a bit confusing. I'd remove the dry-run from the confirmation function and just:
```python
if confirmation():
    if not args.dry_run:
           conan_api.remove...
else:
```